Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -67,9 +129,27 @@ private GrpcCopyClient(string host, int grpcPort, bool useCompression)
         /// <param name="useCompression">Whether or not GZip is enabled for copies.</param>
         public static GrpcCopyClient Create(string host, int grpcPort, bool useCompression = false)
         {
-            // TODO: Add caching of GrpcCopyClient objects
-            // TODO: Add case where _clientPool has exceeded some maximum count
-            return new GrpcCopyClient(host, grpcPort, useCompression);
+            if (_clientDict.TryGetValue((host, grpcPort, useCompression), out GrpcCopyClient existingClient))
+            {
+                Interlocked.Increment(ref existingClient._uses);

```

Comment:
Here is another reason why we need a separate type: we should have counters here to track the number of times we reused the clients and the number of times we created new ones.